checkpkg.in

```diff
@@ -85,11 +133,12 @@ for _pkgname in "${pkgname[@]}"; do
 	find-libprovides "$TEMPDIR/$oldpkg" 2>/dev/null | sort > "$TEMPDIR/libraries-$_pkgname-old"
 	find-libprovides "$pkgfile" 2>/dev/null | sort > "$TEMPDIR/libraries-$_pkgname"
 	if ! diff_output="$(sdiff -s "$TEMPDIR/libraries-$_pkgname-old" "$TEMPDIR/libraries-$_pkgname")"; then
-		msg "Sonames differ in $_pkgname!"
+		message="Sonames differ in $_pkgname!"
+		(( WARN )) && warning "$message" || msg "$message"
 		echo "$diff_output"
 	else
 		msg "No soname differences for %s." "$_pkgname"
 	fi
 done
 
-msg "Files saved to %s" "$TEMPDIR"
+(( ! RMDIR )) && msg "Files saved to %s" "$TEMPDIR"
```

Comment:
```suggestion
(( RMDIR )) || msg "Files saved to %s" "$TEMPDIR"
```

If RMDIR is false, this would result in the last command running, having a nonzero return value, thus causing the script to exit nonzero.